internal/handlers/common/compare.go

```diff
@@ -188,15 +189,24 @@ func compare(docValue, filter any) compareResult {
 
 	switch docValue := docValue.(type) {
 	case *types.Document:
+		filter, isFilterDocument := filter.(*types.Document)
```

Comment:
```suggestion
		filter, ok := filter.(*types.Document)
```

"comma ok" is a Go idiom

Reply:
Noted. The lines have been removed. I will handle this case in another PR.